compiler/rustc_transmute/src/lib.rs

```diff
@@ -151,14 +146,7 @@ mod rustc {
             let variant = adt_def.non_enum_variant();
             let fields = match c.try_to_valtree() {
                 Some(ValTree::Branch(branch)) => branch,
-                _ => {
```

Comment:
this happens if the const is still either an infer value or a (potentially generic) unevaluated const?

I would expect us to instead be ambiguous if the `c` is still `Infer` and only return `None`

Reply:
Yeah, I think returning `None` for the ambiguous cases makes sense.